cwf/gateway/tools/uesim_cli/main.go

```diff
@@ -120,25 +132,17 @@ func handleAddUeCmd(cmd *commands.Command, args []string) int {
 		return 1
 	}
 	imsi := strings.TrimSpace(f.Arg(0))
-	if len(authKey) == 0 {
-		fmt.Printf("Subscriber auth_key must be provided as a flag: -auth_key <auth_key>\n")
-		return 1
-	}
-	auth_key, err := hex.DecodeString(authKey)
+
+	ues, err := getConfiguredSubscribers(imsi)
 	if err != nil {
-		fmt.Printf("Could not convert auth key to bytes. Please ensure you've provided auth_key in hex format\n")
-		return 1
+		fmt.Printf("Get UE Error: %s\n", err)
+		return 2
 	}
-	ue, err := createUeConfig(imsi, auth_key, 0)
+	err = addUes(ues)
 	if err != nil {
-		fmt.Printf("Could not create UE config object: %s", err)
+		fmt.Printf("%s\n", err)
 		return 1
 	}
-	err = uesim.AddUE(ue)
```

Comment:
why is addUE now not part of `handleAddUeCmd` ?

Reply:
There was an API getConfiguredSubscribers() in this file in which internally called uesim.AddUE(). This did not seem correct as the use of getConfiguredSubscribers should just be to get the configured subscribers. Hence i added another api addUes which accepts a list of UEs  to be added to uesim. 
The same function addUes() is now called from this function which  encapsulates uesim.AddUE()